### DISCUSSION TITLE
return icon only on BadgeColumn

### DISCUSSION THREAD

**Discussion by aleirsyad** [filamentphp/filament#8264] [Help] [answered]
1. i have a model with these column (id, user_id, subject, message, and reply)
2. here's the badgecolumn code on the columns field

```
BadgeColumn::make('reply')
                    ->label('Status')
                    ->icon(static function ($state): string {
                        if ($state != '') {
                            return 'heroicon-o-check';
                        } 
                            return 'heroicon-o-x';
                        
                    })
                    ->color(static function ($state): string {
                        if ($state != '') {
                            return 'success';
                        }
                        
                        return 'secondary';
                    }), 
```
3. i want to return only icon and still struggle with the conditional closure because when the reply still null there's no return

here's the image 

![image](https://github.com/filamentphp/filament/assets/91279774/9dd9063f-7043-47f5-a0b1-d78f1819df2b)
![image](https://github.com/filamentphp/filament/assets/91279774/bcfa22de-49a8-4e75-86c9-3b6d3107fa10)

**Comment by leandrocfe** [comment]
[Icon Column](https://filamentphp.com/docs/3.x/tables/columns/icon) ?

**Comment by risangbaskoro** [comment]
If you're doing it in v2, you're in the correct path. I assume that you want to have a badge that show text with a icon prefix, conditionally.

What I would do instead is change the closure of the icon and color. Try using a non-static function.

Also, I would recommend using `!is_null($state)` instead of `$state != ''` for readability. Cheers!

**Comment by aleirsyad** [comment, marked as answer]
Thanks for the answer and now I already found the solution. I also use formatStateUsing just like this

```
BadgeColumn::make('reply')
                    ->label('Status')
                    ->icon(function ($state): string {
                        if (!is_null($state)) {
                            return 'heroicon-o-check';
                        }
                        return 'heroicon-o-x';
                    })
                    ->color(function ($state): string {
                        if (!is_null($state)) {
                            return 'success';
                        }

                        return 'danger';
                    })
                    ->formatStateUsing(function ($state): string {
                        if (!is_null($state)) {
                            return 'Sudah Dijawab';
                        }
                        return 'Belum Terjawab';
                    }),
```

![image](https://github.com/filamentphp/filament/assets/91279774/93d28813-ac49-48fd-b1f0-08dbf467347e)
